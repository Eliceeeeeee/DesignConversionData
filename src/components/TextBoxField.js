import React from 'react';
import $ from 'jquery'

class TextBoxField extends React.Component {
	constructor(props) {
        super(props);
        this.setInputValue = this.setInputValue.bind(this);
	}
    componentDidMount() {
		this.setInputValue(this.props.value);
	}
	componentWillReceiveProps(nextProps) {
        this.setInputValue(nextProps.value);
    }
    setInputValue(value) {
		if (!isNaN(value) || value) {
			$('#'+this.props.id).val(value).change();
		}
	}
    render() {
        return (
            <input type="text"
                id={this.props.id}
                minLength= {this.props.minLength}
                maxLength={this.props.maxLength}
                className={this.props.className}
                onChange={this.props.onChange}
            />
        );
    }
}

export default TextBoxField;