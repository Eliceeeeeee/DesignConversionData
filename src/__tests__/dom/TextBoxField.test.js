import React from 'react';
import { shallow } from 'enzyme';
import TextBoxField from '../../components/TextBoxField';

describe('TextBoxField should match the snapshot with conditions:', () => {
    it('has no props', () => {
        const component = shallow(<TextBoxField />);
        expect(component).toMatchSnapshot();
    });

    it('has id props', () => {
        const component = shallow(<TextBoxField id= "input-paging-no" />);
        expect(component).toMatchSnapshot();
    });

    it('has minLength props', () => {
        const component = shallow(<TextBoxField minLength="1" />);
        expect(component).toMatchSnapshot();
    });

    it('has maxLength props', () => {
        const component = shallow(<TextBoxField maxLength="2" />);
        expect(component).toMatchSnapshot();
    });

    it('has className props', () => {
        const component = shallow(<TextBoxField className="input-text-field" />);
        expect(component).toMatchSnapshot();
    });

    it('has value props and when receive new props', () => {
        let newValue = '1';
        const component = shallow(<TextBoxField value={newValue} />);
        expect(component).toMatchSnapshot();

        newValue = '3';
        component.setProps({ newValue });
        expect(component).toMatchSnapshot();
    });

    it('has value props', () => {
        const component = shallow(<TextBoxField value="3" />);
        expect(component).toMatchSnapshot();
    });
});
