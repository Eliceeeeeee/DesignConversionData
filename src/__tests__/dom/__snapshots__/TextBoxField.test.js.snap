// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBoxField should match the snapshot with conditions: has className props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    className="input-text-field"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "input-text-field",
      "id": undefined,
      "maxLength": undefined,
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "input-text-field",
        "id": undefined,
        "maxLength": undefined,
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has id props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    id="input-paging-no"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": "input-paging-no",
      "maxLength": undefined,
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": "input-paging-no",
        "maxLength": undefined,
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has maxLength props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    maxLength="2"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": undefined,
      "maxLength": "2",
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": undefined,
        "maxLength": "2",
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has minLength props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    minLength="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": undefined,
      "maxLength": undefined,
      "minLength": "1",
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": undefined,
        "maxLength": undefined,
        "minLength": "1",
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": undefined,
      "maxLength": undefined,
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": undefined,
        "maxLength": undefined,
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has value props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    value="3"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": undefined,
      "maxLength": undefined,
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": undefined,
        "maxLength": undefined,
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has value props and when receive new props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    value="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": undefined,
      "maxLength": undefined,
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": undefined,
        "maxLength": undefined,
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TextBoxField should match the snapshot with conditions: has value props and when receive new props 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextBoxField
    newValue="3"
    value="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "id": undefined,
      "maxLength": undefined,
      "minLength": undefined,
      "onChange": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "id": undefined,
        "maxLength": undefined,
        "minLength": undefined,
        "onChange": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
